text/0000-deployment-language.md

```diff
@@ -0,0 +1,561 @@
+
+Troy files can be deployed to a Tremor instance or cluster on startup or via API endpoint. It is also possible to issue Troy statements inside a REPL-like setup, but this is outside of the scope of this RFC.
+
+Whenever a Troy file is deployed the statements within it are executed.
+
+Similar to trickle, Troy also supports modularization and `use` statements using the same mechanism as is employed in trickle (loading the "`used`" file contents into the current file). This allows for defining commonly used artefacts in separate files.
+
+## Example
+
+```
+
+# define a connector with arguments
+define http_server connector my_http_connector
+args
+  server = "Tremor/1.2.3",
+with
+  codec = "json",
+  interceptors = [ "newline" ],
+  config.headers = {
+      "Server": args.server
+  }
+end;
+
+# define a file connector
+define file connector my_file
+with
+    config.path = "/snot/badger.json",
+    codec = "json",
+    interceptors = [ "newline" ]
+end;
+
+# define the simplest pipeline
+define pipeline passthrough
+query
+    select event from in into out;
+end;
+
+# instantiate a connector
+create connector instance "01" from my_http_connector
+with
+    server = "Tremor/3.2.1"
+end;
+
+# define a flow - connecting artefact instances
+define flow my_flow
+links
+    # reference artefacts by id - these instances are not recreated if they are already created
+    connect "/connector/my_http_connector/01:out" to "/pipeline/passthrough:in";
+    # implicit default ports - here we auto-create instances
+    connect passthrough to my_file;
+end;
+
+# instantiate a flow - let the events flow ~~~
+create instance "01" from flow/my_flow;
+```
+
+
+
+Explain the proposal as if it was already included in Tremor and you were teaching it to another Tremor user. That generally means:
+
+- Introducing new named concepts.
+- Explaining the feature largely in terms of examples.
+- Explaining how stakeholders should *think* about the feature, and how it should impact the way they use tremor. It should explain the impact as concretely as possible.
+- If applicable, provide sample error messages, deprecation warnings, or migration guidance.
+- If applicable, describe the differences between teaching this to existing tremor stakeholders and new tremor programmers.
+
+For implementation-oriented RFCs (e.g. for language internals), this section should focus on how language contributors should think about the change, and give examples of its concrete impact. For policy RFCs, this section should provide an example-driven introduction to the policy, and explain its impact in concrete terms.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+Troy supports 2 very basic operations / kinds of statements:
+
+ * Definition of artefacts
+ * Creation of artefacts
+
+## Artefact definition
+
+Artefact definition creates a referencable artefact inside the tremor repository.
+It contains:
+
+ * an id that is unique per
```

Comment:
is it worth mentioning `unique per cluster` here?

Reply:
gonna write `per repository` then we are on the safe side once we extend the repository to the whole cluster.